docs/source/filter_containers.rst

```diff
@@ -0,0 +1,122 @@
+.. _filter_containers:
+
+=================
+Filter containers
+=================
+
+A filter container is just a regular python container that has some extra methods for filtering or ordering its
+elements. It has the same interface (i.e. attributes and methods) as the primitive python type it inherits from, with
+these extra methods:
+
+- ``filter``
+- ``order_by``
+
+There are two types of filter containers currently implemented:
+
+- ``FilterSet``
+- ``FilterList``
+
+
```

Comment:
The following example leaps directly into `FilterSet` which is great, and you've got the example above for how to filter datasets. But it'd be good to have:
- A cross reference ('for example, the :ref:``files`` attribute of a Dataset is filterable<crossref>>_.')
- A list of examples of where FilterSets are used (currently DataSet.files and .tags attributes of Dataset, DataFile and some other things??)